app/controllers/application_controller.rb

```diff
@@ -4,8 +4,8 @@ class ApplicationController < ActionController::Base
   include GDS::SSO::ControllerMethods unless Settings.basic_auth.enabled
   include Pundit::Authorization
   before_action :set_request_id
-  before_action :authenticate
   before_action :check_service_unavailable
+  before_action :authenticate
```

Comment:
I'm really not a huge fan of having loads of before_actions, especially if the order is important. Could we maybe create a private method in this controller to merge some of these and then have a test for the order that they called?

Reply:
I put the `authenticate` and `check_access` before actions together in one method. I don't think we can put any other before actions together, as we need to be able to skip different actions in different controllers.